ntp-daemon/src/peer.rs

```diff
@@ -47,21 +47,16 @@ pub async fn start_peer<A: ToSocketAddrs, C: 'static + NtpClock + Send>(
                 },
                 result = socket.recv(&mut buf) => {
                     // Note: packets are allowed to be bigger when including extensions.
-                    // we don't expect them, but the server may still send them. They
-                    // are guaranteed safe to ignore.
-                    if let Ok((size, Some(timestamp))) = result {
-                        if size >= 48 {
-                            let packet = NtpHeader::deserialize(&buf);
+                    // we don't expect them, but the server may still send them. The
+                    // extra bytes are guaranteed safe to ignore. `recv` truncats the messages
+                    if let Ok((_, Some(timestamp))) = result {
```

Comment:
Switch the order so we can early continue? Can also pull the `tx.send()` out of the `if` expression.

Reply:
I am not sure what you suggest here with switch the order. Also, pulling out the tx.send out of the two if expressions at the end is not really an option, as they don't have an else branch (there is an implicit no-update else there, which semantically matters because calling tx.send itself is the trigger for further updates which are not always needed)